apps/emqx_durable_storage/IMPLEMENTATION.md

```diff
@@ -3,9 +3,17 @@
 In the logic layer we don't speak about replication.
 This is because we could use an external DB with its own replication logic.
 
-On the other hand, we introduce notion of shard right here at the logic.
+On the other hand, we introduce notion of shard right here at the logic layer.
 This is because shared subscription logic needs to be aware of it to some extend, as it has to split work between subscribers somehow.
 
+
+# Modus operandi
```

Comment:
Suggest to add 'First delivery' section with the following content (below). Based on my meetings notes from the first part of the session about first delivery scope, processed with ChatGPT (as you could tell).

> The recent efforts to integrate RocksDB into our system are primarily aimed at improving the existing mechanism, which is based on ETS tables. The shortcoming of the current mechanism lies in its incapability to preserve messages when a node goes offline, as it is entirely RAM-based. Additionally, its high demand for RAM significantly restricts scalability.
> In the past, we made attempts to use Mnesia as a possible alternative. However, the performance test did not yield promising results, leading us to explore other alternatives. Our ultimate objective, however, is not to devise a flawless solution akin to systems such as Kafka. Instead, our focus is on delivering an improved solution compared to previous iterations.
> The key features we aim to implement include durability, scalability, and an adequate speed. The first phase of the delivery will not incorporate a replication feature, implying that there will only be a single copy of messages in the cluster. Consequently, if a node goes offline, we would temporarily lose access to the messages stored on that node. However, these messages will be recoverable once the node is back online.
While the initial version of this integration may not feature replication, we are designing it with a foresight for future replication capabilities. We are optimistic that this new integration will significantly elevate the efficiency and reliability of our system.

Reply:
nvm, added to Confluence